### PR TITLE
entrypoint.sh:wait for mod to finish building before testing existence

### LINKED ISSUE #5: entrypoint.sh does not wait for packaging to finish
On bigger mods packaging can take more than a few seconds. See https://github.com/0ADMods/xiiiad/actions/runs/3224966299/jobs/5276698900

Currently the script doesn't wait for it to finish, which can lead to a race condition

```sh
su user0ad --command "/home/user0ad/usr/bin/pyrogenesis  \
    -mod=package_mod   \
    -archivebuild=$INPUT_DIRECTORY  \
    -archivebuild-output=$OUTPUT_FILE    \
    -archivebuild-compress"

test -f "$OUTPUT_FILE"
zip -d "$OUTPUT_FILE" ".git*"
```

Should maybe be
```sh
su user0ad --command "/home/user0ad/usr/bin/pyrogenesis  \
    -mod=package_mod   \
    -archivebuild=$INPUT_DIRECTORY  \
    -archivebuild-output=$OUTPUT_FILE    \
    -archivebuild-compress" && \
    test -f "$OUTPUT_FILE"  && \
   zip -d "$OUTPUT_FILE" ".git*"

```
And find a way to wait for that.

### PR DESCRIPTION
**Pull request by andy5995**
Also add a test

fixes #5

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -29,14 +27,23 @@ jobs:
     - uses: actions/checkout@v3
       with:
         submodules: true
-    - uses: 0ad-matters/gh-action-build-pyromod@trunk
+    - uses: 0ad-matters/gh-action-build-pyromod@delay
```
Comment (andy5995) [inline .github/workflows/test.yml]: Trying to find out how to set the correct branch name for this...

https://github.com/orgs/community/discussions/35812

**Comment by StanleySweet** [comment]
I wondering why it didn't wait in the first place. Maybe because it's ran as another user?

**Comment by andy5995** [comment]
> I wondering why it didn't wait in the first place. Maybe because it's ran as another user?

Looks like you're using a slightly outdated version of the example.

Note in https://github.com/0ADMods/xiiiad/actions/runs/3224966299/jobs/5276698900#step:5:28 it's using github.repository for the output filename. That would be 0ADMods/xiiiad. In prior examples, the example cut the leading 'owner/' from the MOD_NAME variable.

I think I've updated the examples everywhere, but maybe I'm wrong. Where did you get the one you're using from?

**Comment by andy5995** [comment]
@StanleySweet When I run this locally all as one command, there's no error

`{ ~/Downloads/0ad-0.0.26-alpha-2210062307-x86_64.AppImage      -mod=package_mod       -archivebuild=xiiiad      -archivebuild-output=xiiiad-test.pyromod        -archivebuild-compress;zip -d xiiiad-test.pyromod ".git*"; }`

**Comment by StanleySweet** [comment]
I went to look for it  in the history because I wanted it to be dynamic in case I ever renamed the repo. Sucks that it includes the organisation name :/

**Comment by andy5995** [comment]
https://github.com/0ad-matters/gh-action-build-pyromod/commit/0a542da3dd05282fd49f209041a26c1e904d4d33#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R58

**Comment by andy5995** [comment]
> [0a542da#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R58](https://github.com/0ad-matters/gh-action-build-pyromod/commit/0a542da3dd05282fd49f209041a26c1e904d4d33#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R58)

@StanleySweet Another way to get the mod name dynamically is use the command-line json parser (not available by default on all Linux distros)

`MOD_NAME="$(jq .name mod.json | tr -d \")"`

**Pull request merged**